### PR TITLE
chore: use kqueue backend of notify on macOS

### LINKED ISSUE #20996: Zero macOS framework dependencies
Action items:

- [x] lazy load Security.framework #18071 
  - [x] #20824 
  - [x] #18072 
- [x] replace all chrono::Utc::now() usage with `cli::time::utc_now()`. 
  - [x] #18088 
  - [x] https://github.com/denoland/deno/pull/20995
- [x] update `nextest/quick-junit` to not use Chrono's "clock" feature. 
  - [x] https://github.com/nextest-rs/nextest/pull/1073
  - [x] https://github.com/denoland/deno/pull/21004
- [x] https://github.com/denoland/deno/pull/21027
- [ ] https://github.com/denoland/deno/pull/21028
  ```toml
  # Note: Do not use the "macos_fsevent" feature of notify, as it links us to CoreFoundation on macOS.
  notify = { version = "=5.0.0", default-features = false, features = ["macos_kqueue"] }
  ```

### PR DESCRIPTION
**Pull request by littledivy**
Fixes #20996 

"macos_fsevent" feature of notify links us to CoreFoundation on macOS.

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Is there a difference in performance?

**Comment by littledivy** [comment]
Hard to tell until all of the PRs land. This one needs review: https://github.com/denoland/deno/pull/18072

**Review by bartlomieju — APPROVED** [review]

File: cli/tests/unit/fs_events_test.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ async function getTwoEvents(
   const events = [];
   for await (const event of iter) {
     events.push(event);
-    if (events.length > 2) break;
+    if (events.length >= 2) break;
```
Comment (mmastrac) [inline cli/tests/unit/fs_events_test.ts:35]: Nit: leave a comment here describing why it might be > 2 in some cases.

**Review by mmastrac — APPROVED** [review]
LGTM one nit

**Pull request merged**